### PR TITLE
Run g-w linker script before ctest for prepoceanobs task

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
In anticipation of the merging of https://github.com/NOAA-EMC/global-workflow/pull/2459 , the `test_gdasapp_soca_setup_obsprep` ctest which sets up for `test_gdasapp_soca_JGLOBAL_PREP_OCEAN_OBS` now runs the global-workflow `sorc/link_workflow.sh` script to allow the observation config files to be accessed in a global-workflow-compliant way. 

Effectively addresses https://github.com/NOAA-EMC/GDASApp/issues/1032

### CODE REVIEW

File: test/soca/gw/setup_obsprep.sh [hunk 1]
```diff
@@ -13,6 +13,11 @@ project_source_dir="$1"
 testdatadir="${project_source_dir}/test/soca/testdata"
 testdatadir_bufr="${project_source_dir}/build/gdas/test/testdata"
 
+# run the g-w link script so that the config files can be found
+cd ${project_source_dir}/../../..
```
Comment (guillaumevernieres) [inline test/soca/gw/setup_obsprep.sh]: why not just 
```bash
(cd ${project_source_dir}/../../.. && ./link_workflow.sh)
```
?

**Review by guillaumevernieres — DISMISSED** [review]
Seems fine to me.

File: test/soca/gw/setup_obsprep.sh (hunk 1 above)
Comment (CoryMartin-NOAA) [reply test/soca/gw/setup_obsprep.sh]: Also, can you not do:
`${project_source_dir}/../../../link_workflow.sh` ?

the script should get its path regardless of where it is executed from

**Review by apchoiCMD — DISMISSED** [review]

File: test/soca/gw/setup_obsprep.sh (hunk 1 above)
Comment (AndrewEichmann-NOAA) [reply test/soca/gw/setup_obsprep.sh]: @guillaumevernieres Your bash-fu is superior to mine, Sir
@CoryMartin-NOAA I tried that and for some reason it didn't like it

**Review by apchoiCMD — APPROVED** [review]

**Review by guillaumevernieres — APPROVED** [review]

**Pull request merged**